autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -90,37 +90,35 @@ def __init__(self, path: str, problem_type: str, scheduler_options=None, eval_me
         # self.models_level_all['core'][1] # Stacker level 1
         # self.models_level_all['aux1'][1] # Stacker level 1 aux models, such as weighted_ensemble
         # self.models_level_all['core'][2] # Stacker level 2
+        # Keeps track of the intended use of models in the stack ensemble. When building complex stack ensembles over multiple stages, it can be useful to group related models in the same stack namespace.
+        # This is not used during inference, only during fitting of new models as a convenient way to get base models for a new stack layer.
         self.models_level = defaultdict(dd_list)
 
         self.model_best = None
 
-        self.model_performance = {}  # TODO: Remove in future, use networkx.
-        self.model_paths = {}
-        self.model_types = {}  # Outer type, can be BaggedEnsemble, StackEnsemble (Type that is able to load the model)
-        self.model_types_inner = {}  # Inner type, if Ensemble then it is the type of the inner model (May not be able to load with this type)
-        self.models = {}
-        self.model_graph = nx.DiGraph()
-        self.model_full_dict = {}  # Dict of normal Model -> FULL Model
+        self.models = {}  # Dict of model name -> model object. A key, value pair only exists if a model is persisted in memory.
+        self.model_graph = nx.DiGraph()  # Directed Acyclic Graph (DAG) of model interactions. Contains numerous metadata regarding each model.
```

Comment:
"model interactions" may be less clear than eg: 
"DAG describing how certain models depend on the predictions of other models"

Reply:
added